app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -12,6 +14,7 @@ class MetaRPCClient {
     this.connectionStream.on('data', this.handleResponse.bind(this));
     this.connectionStream.on('end', this.close.bind(this));
     this.responseHandled = {};
+    this.DisconnectError = DisconnectError;
   }
```

Comment:
Just out of curiousity, why do we have to extend `Error` class (creating a new Object), could it be different if we just used `Error` class directly since we are not creating new functions on _Line 6_

Reply:
It's for the use of `instanceof` when checking what error was thrown. 
So that you can differentiate the DisconnectError from an actual error in the operation being performed.